machineconfiguration/v1alpha1/types_internalreleaseimage.go

```diff
@@ -0,0 +1,144 @@
+	// +required
+	Releases []InternalReleaseImageRef `json:"releases,omitempty"`
+}
+
+// InternalReleaseImageRef is used to provide a simple reference for a release
+// bundle. Currently it contains only the name field.
+type InternalReleaseImageRef struct {
+	// name indicates the desired release bundle identifier. This field is required and must be between 1 and 64 characters long.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	Name string `json:"name,omitempty"`
+}
+
+// InternalReleaseImageStatus describes the current state of a InternalReleaseImage.
+type InternalReleaseImageStatus struct {
+	// releases is a list of the release bundles currently owned and managed by the
+	// cluster, indicating that their images can be safely pulled by any cluster entity
+	// requiring them.
+	// Entries must be unique, keyed on the name field.
+	// This field can contain between 1 and 5 entries.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=5
+	// +optional
+	Releases []InternalReleaseImageStatusRef `json:"releases,omitempty"`
+}
+
+type InternalReleaseImageStatusRef struct {
+	// conditions represent the observations of an internal release image current state. Valid types are:
+	// Mounted, Installing, Available, Removing and Degraded.
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=5
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// name indicates the desired release bundle identifier. This field is required and must be between 1 and 64 characters long.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +required
+	Name string `json:"name,omitempty"`
+	// image is an OCP release image referenced by digest.
+	// The format of the image pull spec is: host[:port][/namespace]/name@sha256:<digest>,
+	// where the digest must be 64 characters long, and consist only of lowercase hexadecimal characters, a-f and 0-9.
+	// The length of the whole spec must be between 0 to 447 characters.
+	// The field is optional, and it will be provided after a release will be successfully installed.
+	// +kubebuilder:validation:MinLength=0
+	// +kubebuilder:validation:MaxLength=447
+	// +kubebuilder:validation:XValidation:rule=`self == '' || (self.split('@').size() == 2 && self.split('@')[1].matches('^sha256:[a-f0-9]{64}$'))`,message="the OCI Image reference must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 characters long"
+	// +kubebuilder:validation:XValidation:rule=`self == '' || (self.split('@')[0].matches('^([a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+(:[0-9]{2,5})?/([a-zA-Z0-9-_]{0,61}/)?[a-zA-Z0-9-_.]*?$'))`,message="the OCI Image name should follow the host[:port][/namespace]/name format, resembling a valid URL without the scheme"
+	// +optional
+	Image *string `json:"image,omitempty"`
```

Comment:
@yuqi-zhang @pablintino I reviewed the definition of this field as optional, with a valid empty value, since it will be filled only when a release will get installed (and not, for example, when simply mounted or degraded)

Reply:
Good catch @pablintino - I had initially an issue with the MinLength, which I was able then to resolve by adding the `self == ''`, but then remained at 0. Fixing it